bugbot/rules/inactive_revision.py

```diff
@@ -0,0 +1,328 @@
+                        }
+                        for reviewer in revision["attachments"]["reviewers"][
+                            "reviewers"
+                        ]
+                    ]
+
+                    if any(
+                        reviewer["is_group"] or reviewer["is_accepted"]
+                        for reviewer in reviewers
+                    ) and not any(
+                        not reviewer["is_accepted"]
+                        for reviewer in reviewers
+                        if reviewer["is_blocking"]
+                    ):
+                        continue
+
+                    revisions.append(
+                        {
+                            "rev_id": revision["id"],
+                            "title": revision["fields"]["title"],
+                            "created_at": revision["fields"]["dateCreated"],
+                            "author_phid": revision["fields"]["authorPHID"],
+                            "reviewers": reviewers,
+                        }
+                    )
+
+        user_phids = set()
+        for revision in revisions:
+            user_phids.add(revision["author_phid"])
+            for reviewer in revision["reviewers"]:
+                user_phids.add(reviewer["phid"])
+
+        users = self.user_activity.get_phab_users_with_status(
+            list(user_phids), keep_active=True
+        )
+
+        result: Dict[int, dict] = {}
+        for revision in revisions:
+            author_phid = revision["author_phid"]
+            if author_phid in users:
+                author_info = users[author_phid]
+                revision["author"] = author_info
+            else:
+                continue
+
+            reviewers = []
+            for reviewer in revision["reviewers"]:
+                reviewer_phid = reviewer["phid"]
+                if reviewer_phid in users:
+                    reviewer_info = users[reviewer_phid]
+                    reviewer["info"] = reviewer_info
+                else:
+                    continue
+                reviewers.append(reviewer)
+
+            revision["reviewers"] = [
+                {
+                    "phab_username": reviewer["info"]["phab_username"],
+                }
+                for reviewer in reviewers
+            ]
+            result[revision["rev_id"]] = revision
+
+        return result
+
+    @retry(
+        wait=wait_exponential(min=4),
+        stop=stop_after_attempt(5),
+    )
+    def _fetch_revisions(self, ids: list):
+        return self.phab.request(
+            "differential.revision.search",
+            constraints={"ids": ids},
+            attachments={"reviewers": True},
+        )["data"]
+
+    def _fetch_revision_transactions(self, revision_phid):
+        response = self.phab.request(
+            "transaction.search", objectIdentifier=revision_phid
+        )
+        return response["data"]
+
+    def handle_bug(self, bug, data):
```

Comment:
Instead of querying bugs and then finding revisions attached to the bugs, we could search for revisions directly. WDYT @suhaibmujahid?

Reply:
We will need to query the bugs anyway to ensure we did not comment on them. All the queried bugs have attached revisions. Doing it the other way could be more efficient (not sure), but it will be a bit more complex since `BzCleaner` is designed to start from bugs.